cmd/tailscale/cli/file.go

```diff
@@ -203,7 +203,26 @@ func discoverPeerAPIBase(ctx context.Context, ipStr string) (base string, lastSe
 			return ft.PeerAPIURL, lastSeen, isOffline, nil
 		}
 	}
-	return "", time.Time{}, false, errors.New("target seems to be running an old Tailscale version")
+
+	// Try a bit harder for a nicer error message.
+	found := false
+	if st, err := tailscale.Status(ctx); err == nil && st.Self != nil {
+		for _, peer := range st.Peer {
+			for _, pip := range peer.TailscaleIPs {
+				if pip == ip {
+					found = true
+					if peer.UserID != st.Self.UserID {
+						return "", time.Time{}, false, fmt.Errorf("%s (%s) is owned by a different user; can only send files to your own devices",
+							ipStr, peer.DNSName)
+					}
+				}
+			}
+		}
+	}
+	if found {
+		return "", time.Time{}, false, errors.New("target seems to be running an old Tailscale version")
+	}
+	return "", time.Time{}, false, errors.New("unknown target; not in your Tailnet")
```

Comment:
Is this the first usage of Tailnet in a user-facing error message? Do we have docs anywhere about what a Tailnet is? Maybe s/Tailnet/network/?

Reply:
Suggest s/target /target DNSNAME/ or some other nice identifier, like whatever it is they typed in, so that if there was a typo we are repeating that typo back at them for them to notice.

Reply:
Tailnet is used in the API and existing error messages & flag docs so I think it's safe to use.

I just modified the caller to always include what the they typed as the target in the error message, so these errors can now omit it.